application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -390,64 +369,71 @@ static Optional<TemporaryData> computeTemporaryData(String durationText) {
     }
 
     /**
-     * Wrapper to hold data relevant to temporary actions, for example the time it expires.
+     * Creates a nice looking embed for the mod action taken.
      *
-     * @param expiresAt the time the temporary action expires
-     * @param duration a human-readable text representing the duration of the temporary action, such
-     *        as {@code "1 day"}.
+     * @param guild the guild in which the action has been taken
+     * @param actionTitle the mod action as title e.g, Ban
+     * @param description a short description explaining the action
+     * @param reason reason for the action taken
+     * @param showModmail to show mod contact details or not
+     * @return the embed
      */
-    record TemporaryData(Instant expiresAt, String duration) {
+    static RestAction<EmbedBuilder> getModActionEmbed(Guild guild, String actionTitle,
+            String description, String reason, boolean showModmail) {
```

Comment:
`showModmail` sounds weird. maybe `showModmailAdvice`. and the javadoc then maybe `whether to advice on how to use the modmail command`